### PR TITLE
OSDOCS-2230: Added OpenShift sandboxed containers 4.8 RN

### PR DESCRIPTION
**Pull request by sagidlow**
Preview Links:  

- [OpenShift sandboxed containers](https://deploy-preview-33105--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-8-release-notes.html#ocp-4.8-sandboxed-containers). 
- [Technology Preview](https://deploy-preview-33105--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-8-release-notes.html#ocp-4-8-technology-preview). See the table. I inserted a row with OpenShift sandboxed containers. 
- [Known Issues](https://deploy-preview-33105--osdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-8-release-notes.html#ocp-4-8-known-issues). I added 2 bullets about Known Issues about OpenShift sandboxed containers.

Applies to 4.8 RN. 
https://issues.redhat.com/browse/OSDOCS-2230
QE and SME ack required.

### CODE REVIEW

File: release_notes/ocp-4-8-release-notes.adoc
Comment (jeana-redhat) [inline release_notes/ocp-4-8-release-notes.adoc:224]:
```suggestion
{sandboxed-containers-first} 1.0.0 Technology Preview release introduces built-in support for running Kata Containers as an additional runtime. {sandboxed-containers-first} provide users with the ability to choose from the type of workload that they want to run to cover a wide variety of use cases. You can use the {sandboxed-containers-operator} to perform tasks such as installation and removal, updates, and status monitoring. {sandboxed-containers-first} are only supported on bare metal. {op-system-first} is the only supported operating system for {sandboxed-containers-first} 1.0.0. Disconnected environments are not supported in {product-title} {product-version}.
```

Comment (jeana-redhat) [reply release_notes/ocp-4-8-release-notes.adoc:224]: Might also be worth breaking this blurb into two or more short paragraphs.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (jeana-redhat) [inline release_notes/ocp-4-8-release-notes.adoc:569]: There's not really a rhyme or reason to where things are added to this table, so if you want you could just stick it at the end since it's new. Here is fine too, though :+1:

File: release_notes/ocp-4-8-release-notes.adoc
Comment (jeana-redhat) [inline release_notes/ocp-4-8-release-notes.adoc:673]: s/OSC/(I don't recall what the non-"first" attribute is, but it should go here I think :smiley:)

**Review by jeana-redhat — COMMENTED** [review]
Couple nits, but LGTM. Yay, Release Note blurbs!

File: release_notes/ocp-4-8-release-notes.adoc
Comment (jeana-redhat) [inline release_notes/ocp-4-8-release-notes.adoc:222]: Oops - missed that this also needs an id tag

File: release_notes/ocp-4-8-release-notes.adoc
Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:581]: I'll leave it here for now, unless someone requests it to move it :-)

**Comment by sagidlow** [comment]
@zanetworker, @jensfr, @cpmeadors, @c3d - Please review :-)

File: release_notes/ocp-4-8-release-notes.adoc
Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:688]: @zanetworker - OSC here is intended to be OpenShift sandboxed containers?

File: release_notes/ocp-4-8-release-notes.adoc
Comment (c3d) [inline release_notes/ocp-4-8-release-notes.adoc:237]: "choose from the type of workload", I have a feeling there is a missing word here. Was the intent "choose the runtime based on the type of workload"? In other words, the question for me is "choose what"?

Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:237]: @c3d "Choose the type of workload that they want..."? Maybe drop from?

Comment (c3d) [reply release_notes/ocp-4-8-release-notes.adoc:237]: I believe the same wording issue may exist in #32307

File: release_notes/ocp-4-8-release-notes.adoc
Comment (c3d) [inline release_notes/ocp-4-8-release-notes.adoc:239]: Why does `{product-title} {product-version}` show up as "OpenShift Container Platform Branch Build."? Are we making a support statement about OpenShift or about sandboxed containers?

File: release_notes/ocp-4-8-release-notes.adoc
Comment (c3d) [inline release_notes/ocp-4-8-release-notes.adoc:684]: backtick `hostPath`

File: release_notes/ocp-4-8-release-notes.adoc
Comment (c3d) [reply release_notes/ocp-4-8-release-notes.adoc:237]: @zanetworker @jensfr Were you the ones that came up with the original wording.

It seems to me like we want to explain that we can chose the runtime class on a per-workload basis, but as written, I have no idea what this really says ;-)

File: release_notes/ocp-4-8-release-notes.adoc
Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:239]: Branch Build will be 4.8. It's weird Preview link. When the doc goes live it will say 4.8.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:237]: @c3d nope. The sentence needs to be changed to reflect that we can choose the runtime based on the workload.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:688]: @sagidlow I don't remember writing this, as I never use OSC :)

Maybe it's part of the initial sections in the overview that got moved to the release notes, we kind of were treating the RN section as a draft.

**Comment by zanetworker** [comment]
@sagidlow I think we want to emphasize a bit that sandboxed containers can be consumed via the operator (which is the major addition in this first tech-preview release).

File: release_notes/ocp-4-8-release-notes.adoc
Comment (c3d) [reply release_notes/ocp-4-8-release-notes.adoc:688]: I wrote this part, but I don't recall writing OSC either (maybe I wrote "Kata"?)

File: release_notes/ocp-4-8-release-notes.adoc
Comment (c3d) [reply release_notes/ocp-4-8-release-notes.adoc:237]: @zanetworker I'm confused, because you write "nope" and then repeat what I wrote, except you replace "runtime class on a per-workload basis" with "runtime based on the workload", so you seem to agree with me?

In any case, the way you change the runtime is by selecting a `runtimeClass`, so I believe that we should avoid introducing the notion of "runtime" here. WDYT?

Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:237]: >@zanetworker @jensfr Were you the ones that came up with the original wording.


@c3d the nope was meant to answer your first question.

Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:237]: >In any case, the way you change the runtime is by selecting a runtimeClass, so I believe that we should avoid introducing the notion of "runtime" here. WDYT?

At this point, I believe we should not get to the specific as kubernetes-specific term  vs. a technical reference.

Comment (c3d) [reply release_notes/ocp-4-8-release-notes.adoc:237]: @zanetworker I understand your point. And indeed, we only used "runtime" so far. Can you suggest a wording for @sagidlow to copy-paste?

Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:237]: OpenShift sandboxed containers provide users with the ability to choose Kata Containers as an additional runtime to provide additional isolation for their workloads. The OpenShift sandboxed containers operator is made available to users for performing tasks such as installation, removal, updates, and status monitoring of Kata Containers.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:691]: Was this maybe @jensfr ?

**Comment by sagidlow** [comment]
@c3d, @cpmeadors, @zanetworker, @jensfr - Could I get a second review on the Release Notes? I updated based on feedback.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (cpmeadors) [inline release_notes/ocp-4-8-release-notes.adoc:252]: I am unclear how the operator allows status monitoring of Kata Containers.  I know of no specific feature for this.

**Review by cpmeadors — COMMENTED** [review]
There are additional bugs that may need to be release noted, but I need to see what PM and PO say.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:252]: I don't think it means `status` literally, but rather status (for example, if you describe KataConfig you could see the progress.

**Comment by zanetworker** [comment]
@cpmeadors please flag any issues that are worthy of mention in the release notes.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (c3d) [inline release_notes/ocp-4-8-release-notes.adoc:252]: I watched this in preview mode, it displays as "OpenShift sandboxed containers 1.0.0 Technology Preview". What confuses me is that the only non-capitalized part here is the "most essential" part of the product name.

Could this be changed to show as "OpenShift Sandboxed Containers 1.0.0 Technology Preview"? Or alternatively, show the name in some different style?

Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:252]: @c3d the product name is OpenShift sandboxed containers

**Comment by c3d** [comment]
Thanks @sagidlow, it's looking good (to me at least). Some reservations about the issues we want to mention (@cpmeadors mentioned some that are rearing their ugly head). Hopefully what we see now can be fixed before release. Can we have another commit later with additional issues, or is there a process issue precluding that?

File: release_notes/ocp-4-8-release-notes.adoc
Comment (c3d) [reply release_notes/ocp-4-8-release-notes.adoc:252]: @zanetworker I understand that this is how we currently capitalize it. I still feel the need to point out that the following looks and reads wrong:

_OpenShift sandboxed containers 1.0.0 Technology Preview release introduces built-in support for running Kata Containers as an additional runtime. OpenShift sandboxed containers enables users to choose Kata Containers as an additional runtime to provide additional isolation for their workloads. The OpenShift sandboxed containers Operator enables users to perform tasks such as installation, removal, updates, and status monitoring of Kata Containers._

Maybe simply using a different font, e.g.:

_OpenShift sandboxed containers 1.0.0 Technology Preview release_ introduces built-in support for running _Kata Containers_ as an additional runtime. _OpenShift sandboxed containers_ enables users to choose _Kata Containers_ as an additional runtime to provide additional isolation for their workloads. The _OpenShift sandboxed containers Operator_ enables users to perform tasks such as installation, removal, updates, and status monitoring of _Kata Containers_.

Notice how the Parts in italicS Are totally Inconsistent in Their capitAlization.

Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:252]: @c3d Discussions about this has happened previously. The approved product name is OpenShift sandboxed containers. Per our guidelines we have to capitalize our Operators. However, the approved product name is OpenShift sandboxed containers and OpenShift sandboxed containers Operator. Kata Containers is what I have found in their documentation in terms. It looks weird definitely, but this is what has been approved.

Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:252]: > The OpenShift sandboxed containers Operator enables users to perform tasks such as installation, removal, updates, and monitoring status using the `KataConfig` custom resource.

**Comment by sagidlow** [comment]
> Thanks @sagidlow, it's looking good (to me at least). Some reservations about the issues we want to mention (@cpmeadors mentioned some that are rearing their ugly head). Hopefully what we see now can be fixed before release. Can we have another commit later with additional issues, or is there a process issue precluding that?

@c3d - We can definitely do another PR (if needed) with additional issues. We do have to have all content in RN, reviewed, and committed by end of June. We have to hand off to translation on July 1.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:252]: Which documentation are you referring to @sagidlow. I think Christoph's suggestion is trying to distinguish usage visually using italics as an example.

Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:252]: Suggestion: The OpenShift sandboxed containers Operator enables users to perform tasks such as installation, removal, updates, and monitoring installation state visible when describing the KataConfig custom resource.

Comment (c3d) [reply release_notes/ocp-4-8-release-notes.adoc:252]: @zanetworker I suspect that @sagidlow is referring to how Kata Containers is spelled on their web site, e.g. https://katacontainers.io. It is rather consistently spelled with two capitals.

Frankly, even if that's something that needs to be revisited at the product management level, I think it's worth bringing up the issue that the current capitalisation makes documentation unreadable. In particular if we start capitalizing "operator"! It seriously takes me three or four back and forth scans in a sentence to simply be able to parse "OpenShift sandboxed containers 1.0.0 Technology Preview release introduces built-in support". To me, this is bad enough to turn into a UX experience at this point.

Different fonts could definitely help. Or the use of well-defined acronyms. If those are not options, then we need to pay attention to what the phrasing looks like in the end, something that may be a bit obscured to the writers because of the use of macros. For example 

_The release of the Technical Preview for OpenShift sandboxed containers 1.0.0 introduces built-in support for running Kata Containers as an additional runtime_

Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc]: @zanetworker @c3d - With our current guidelines we cannot use italics in this manner. We also cannot implement different fonts, and so on. 

We can talk about changing the name of the Operator or the product itself by using proper capitalization. But unfortunately, unless we can fix the product name. As @c3d this is def. a UX experience issue and if we can change OpenShift sandboxed containers to OpenShift Sandboxed Containers that would make more sense and more align with how our products are titled.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (zanetworker) [inline release_notes/ocp-4-8-release-notes.adoc:532]: The version appears as `Branch Build` sometimes.

Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:532]: @zanetworker `Branch Build` will result in the 4.8 product version showing up.

**Comment by jensfr** [comment]
lgtm 
I'm aware that there are more bugs that we might have to add, but the discussion is still going on. As it looks now it looks good to me.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (zanetworker) [inline release_notes/ocp-4-8-release-notes.adoc:379]:
```suggestion
{sandboxed-containers-first} 1.0.0 Technology Preview release introduces built-in support for running Kata Containers as an additional runtime. {sandboxed-containers-first} enables users to choose Kata Containers as an additional runtime to provide additional isolation for their workloads. The {sandboxed-containers-operator} automates the tasks of installing, removing, and updating Kata Containers. It allows for tracking the state of those tasks by describing the `KataConfig` custom resource.
```

Comment (zanetworker) [reply release_notes/ocp-4-8-release-notes.adoc:379]: One of the goals of the operator is to offload those tasks from the users.

**Comment by zanetworker** [comment]
@sagidlow a small change suggested by me, otherwise LGTM

**Comment by cpmeadors** [comment]
> @cpmeadors please flag any issues that are worthy of mention in the release notes.

Sorry I didn't seen this sooner.  What process are we using to flag release note issues?

**Comment by sagidlow** [comment]
> > @cpmeadors please flag any issues that are worthy of mention in the release notes.
> 
> @sagidlow Sorry I didn't seen this sooner. What process are we using to flag release note issues?

Do you mean things that should be a part of the Known Issues?

**Comment by cpmeadors** [comment]
> > > @cpmeadors please flag any issues that are worthy of mention in the release notes.
> > 
> > 
> > @sagidlow Sorry I didn't seen this sooner. What process are we using to flag release note issues?
> 
> @cpmeadors Do you mean things that should be a part of the Known Issues?

@sagidlow Yes. How would I flag an issue to appear in the release notes as a known issue?

**Comment by vikram-redhat** [comment]
@sagidlow is this ready to be merged?

File: release_notes/ocp-4-8-release-notes.adoc
Comment (cpmeadors) [inline release_notes/ocp-4-8-release-notes.adoc:1094]: The reason for this to be included in the known issues was to reduce questions/support requests about completeness/readiness of the operator.  It is Tech Preview, but the incomplete information could indicate that it is unfinished more that to Tech Preview standards.  I don't know how much we have to explain, but that is why it is here.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (cpmeadors) [inline release_notes/ocp-4-8-release-notes.adoc:1098]: Technically, there is no failure here as sandboxed-containers-operator is unique (I think) in limiting the number of CRs created.  The issue here is how we handle a user creating additional CRs and setting correct expectations, Currently, people may assume they can create multiple CRs and then get confused when nothing changes.  Again, just background on the issue.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (cpmeadors) [inline release_notes/ocp-4-8-release-notes.adoc:1090]: This is just a feature that was not tested to our satisfaction.  It may work in some cases, but not all.  Just trying to set expectations.

**Review by cpmeadors — COMMENTED** [review]
Just added a bit of context around the issues.  Use what you like.  I am not sure how much explanation is needed.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:1098]: @cpmeadors Should I clarify with something like this?

> Creating multiple `KataConfig` custom resources results in a silent failure.  The console does not provide a prompt to notify the user that creating more than one custom resource has failed. 

Let me know if this is not what you're looking for.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:1094]: @cpmeadors In that case, I would clarify with something like this?

> The {sandboxed-containers-operator} details page on Operator Hub contains a few missing fields. The missing fields do not prevent you from installing the {sandboxed-containers-operator} in 4.8.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (sagidlow) [reply release_notes/ocp-4-8-release-notes.adoc:1090]: @cpmeadors How about something like?

> In the 4.8 release, the {sandboxed-containers-operator} does not handle the custom `MachineConfigPool`. Attempting to use customized `MachineConfigPool` may result in consistent behavior while installing and uninstalling the Operator. 

I try to steer clear of saying, well it might work in this instance but not in this...not sure. Any thoughts?

File: release_notes/ocp-4-8-release-notes.adoc
Comment (cpmeadors) [reply release_notes/ocp-4-8-release-notes.adoc:1094]: Works for me.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (cpmeadors) [reply release_notes/ocp-4-8-release-notes.adoc:1098]: I like that wording.

File: release_notes/ocp-4-8-release-notes.adoc
Comment (cpmeadors) [reply release_notes/ocp-4-8-release-notes.adoc:1090]: My thoughts exactly.  Looks good.

**Comment by sagidlow** [comment]
@zanetworker, @cpmeadors , @c3d, @jensfr - Ready for last pass before merge. When I sent out to the kata-team no one had any comments. Please provide final comments by end of day, local time, **Friday, June 25**.

**Review by cpmeadors — APPROVED** [review]
Everything looks good.

**Comment by ahardin-rh** [comment]
LGTM! My only question: When we reference the Operator itself should it be a proper noun? Should it be OpenShift Sandboxed Containers Operator instead of OpenShift sandboxed containers Operator?

File: release_notes/ocp-4-8-release-notes.adoc
Comment (jensfr) [inline release_notes/ocp-4-8-release-notes.adoc:1097]: The problem here is that the actual feature (selecting nodes for the installation by adding a label to it) does work, but the UI that we present to the user is wrong. It can be worked around by adding the kataConfigPoolSelector manually. I suggest to add:

The impact of this is that the scheduling.nodeSelector will not be set, which means Pods using the
runtime class 'kata' could be scheduled to Nodes that don’t have the Kata Containers runtime installed.
This problem can be worked around by specifying the kataConfigPoolSelector manually in the kataconfig. In a terminal do 'oc edit kataconfig example-kataconfig' then add the selector.  A full example of a kataconfig CR

```
apiVersion: kataconfiguration.openshift.io/v1
kind: KataConfig
metadata:
  name: example-kataconfig
  spec:
    kataConfigPoolSelector:
      matchLabels:
        custom-kata1: test
```

File: release_notes/ocp-4-8-release-notes.adoc
Comment (jeana-redhat) [inline release_notes/ocp-4-8-release-notes.adoc:1321]:
```suggestion
* In the 4.8 release, adding a value to `kataConfgPoolSelector` by using the {product-title} web console causes `scheduling.nodeSelector` to be populated with an empty value. Pods that use `RuntimeClass` with the value of `kata` might be scheduled to nodes that do not have the Kata Containers runtime installed.
```

File: release_notes/ocp-4-8-release-notes.adoc
Comment (jeana-redhat) [inline release_notes/ocp-4-8-release-notes.adoc:1323]:
```suggestion
To work around this issue, specify the `nodeSelector` value manually in the `RuntimeClass` `kata` by running the following command:
```
just trying to avoid "by [gerund] by [gerund]" here :)

**Review by jeana-redhat — COMMENTED** [review]
Minor editorial suggestions, yay for finally being through with this one! :)

**Pull request merged**
